tomviz/PythonGeneratedDatasetReaction.cxx

```diff
@@ -129,33 +130,35 @@ class PythonGeneratedDataSource : public QObject
       result = this->MakeDatasetFunction.call(args, kwargs);
       if (!result.isValid()) {
         qCritical() << "Failed to execute script.";
-        return retVal;
+        return nullptr;
       }
     }
 
-    vtkSMSessionProxyManager* pxm =
-      tomviz::ActiveObjects::instance().proxyManager();
-    vtkSmartPointer<vtkSMProxy> source;
-    source.TakeReference(pxm->NewProxy("sources", "TrivialProducer"));
-    vtkTrivialProducer* tp =
-      vtkTrivialProducer::SafeDownCast(source->GetClientSideObject());
-    tp->SetOutput(image.Get());
-    source->SetAnnotation(tomviz::Attributes::TYPE, "DataSource");
-    source->SetAnnotation("tomviz.DataSource.FileName",
-                          "Python Generated Data");
-    source->SetAnnotation(tomviz::Attributes::LABEL,
-                          this->label.toLatin1().data());
-    source->SetAnnotation("tomviz.Python_Source.Script",
-                          this->pythonScript.toLatin1().data());
-    source->SetAnnotation("tomviz.Python_Source.X",
-                          QString::number(shape[0]).toLatin1().data());
-    source->SetAnnotation("tomviz.Python_Source.Y",
-                          QString::number(shape[1]).toLatin1().data());
-    source->SetAnnotation("tomviz.Python_Source.Z",
-                          QString::number(shape[2]).toLatin1().data());
-
-    retVal = vtkSMSourceProxy::SafeDownCast(source.Get());
-    return retVal;
+    QVariantMap vmap;
+    foreach (QString key, this->arguments.keys()) {
+      vmap[key] = this->arguments[key];
+    }
+
+    QJsonObject argsJson = QJsonObject::fromVariantMap(vmap);
+
+    QJsonArray size;
+    size.append(shape[0]);
+    size.append(shape[1]);
+    size.append(shape[2]);
+
+    QJsonObject description;
+    description["pythonScript"] = this->pythonScript;
```

Comment:
Let's just call it script, this was used in the python operator too, I would argue that the member variable should be renamed too, but fine leaving as-is